src/structures/StageChannel.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+const GuildChannel = require('./GuildChannel');
+const Collection = require('../util/Collection');
+const Permissions = require('../util/Permissions');
+
+/**
+ * Represents a guild stage channel on Discord.
+ * @extends {GuildChannel}
+ */
+class StageChannel extends GuildChannel {
+  _patch(data) {
+    super._patch(data);
+
+    /**
+     * The topic of the stage channel
+     * @type {?string}
+     */
+    this.topic = data.topic;
+
+    /**
+     * The RTC region for this stage channel. This region is automatically selected if `null`.
+     * @type {?string}
+     */
+    this.rtcRegion = data.rtc_region;
+
+    /**
+     * The bitrate of this stage channel
+     * @type {number}
+     */
+    this.bitrate = data.bitrate;
+
+    /**
+     * The maximum amount of users allowed in this channel.
+     * @type {number}
+     */
+    this.userLimit = data.user_limit;
+  }
+
+  /**
+   * The members in this stage channel
+   * @type {Collection<Snowflake, GuildMember>}
+   * @readonly
+   */
+  get members() {
+    const coll = new Collection();
+    for (const state of this.guild.voiceStates.cache.values()) {
+      if (state.channelID === this.id && state.member) {
+        coll.set(state.id, state.member);
+      }
+    }
+    return coll;
+  }
+
+  /**
+   * Checks if the stage channel is full
+   * @type {boolean}
+   * @readonly
+   */
+  get full() {
+    return this.userLimit > 0 && this.members.size >= this.userLimit;
+  }
+
+  /**
+   * Whether the channel is deletable by the client user
+   * @type {boolean}
+   * @readonly
+   */
+  get deletable() {
+    return this.manageable && this.permissionsFor(this.client.user).has(Permissions.FLAGS.CONNECT, false);
+  }
+
+  /**
+   * Whether the channel is editable by the client user
+   * @type {boolean}
+   * @readonly
+   */
+  get editable() {
+    return this.manageable && this.permissionsFor(this.client.user).has(Permissions.FLAGS.CONNECT, false);
+  }
```

Comment:
Why not merge these 2 into one `manageable` property? They're literally the same